astropy/tests/runner.py

```diff
@@ -12,10 +12,10 @@
 from importlib.util import find_spec
 from pathlib import Path
 
-from astropy.utils import find_current_module
+from astropy.utils import deprecated, find_current_module
 from astropy.utils.exceptions import AstropyDeprecationWarning, AstropyWarning
 
-__all__ = ["TestRunner", "TestRunnerBase", "keyword"]
+__all__ = ["TestRunner", "TestRunnerBase"]
```

Comment:
`keyword` itself is a decorator, so it is weird to put a `@deprecated` decorator on a decorator. Not even sure why it is exposed in `__all__`. I think probably unintentional in the early days of `astropy`. Much easier to take it off public API. I don't think downstream should be using it on its own anyway.

Reply:
> so it is weird to put a @deprecated decorator on a decorator

I disagree, there's technically nothing wrong with decorating a decorator. The fact that it feels weird (it does !) isn't sufficient reason to remove public API without a warning. We've had enough problems with removing stuff that was explicitly private, I don't fancy the idea of knowingly removing stuff that's explicitly public :/

Reply:
I still don't like decorator on a decorator. Let's wait for others and see. Thanks!

Reply:
hum, no other voice was raised, so I'm not sure how we can make a decision here. If it helps, I'm willing to let my suggestion ignored (assuming there are no known usage of `keyword` downstream), but we need to keep an eye out for sudden breakage and be ready to revert this bit to smooth out the transition.